app/controllers/api/base_controller.rb

```diff
@@ -19,16 +47,62 @@ def respond_with_conflict(json_hash)
 
     # Use logged in user (spree_current_user) for API authentication (current_api_user)
     def authenticate_user
-      @current_api_user = try_spree_current_user
-      super
+      return if @current_api_user = try_spree_current_user
```

Comment:
Is this really equivalent? previously, we called `super`, which means that the parent's logic was always executed. Is the logic below not a copy&paste from that?

Reply:
It looks equivalent logically. :+1: When executing `super`, the method is exited if `@current_api_user` is truthy, which is what happens here too.